src/Microsoft.Azure.WebJobs.Extensions.Kafka/Diagnostics/ActivityProvider.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Host.Executors;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka.Diagnostics
+{
+    internal class ActivityProvider
+    {
+        public static readonly ActivitySource ActivitySource = new ActivitySource("Microsoft.Azure.Webjobs.Extensions.Kafka");
```

Comment:
ActivitySource instance is being instantiating at multiple places and this object is being accessed in the implemented classes. I would recommend make a Singleton class and access ActivitySource from there

Reply:
The type of ActivitySource is static, so it would be instantiated only once. Yes, it is being accessed at multiple places.